clang/lib/Sema/SemaType.cpp

```diff
@@ -8612,7 +8612,10 @@ static void HandleLifetimeBoundAttr(TypeProcessingState &State,
     CurType = State.getAttributedType(
         createSimpleAttr<LifetimeBoundAttr>(State.getSema().Context, Attr),
         CurType, CurType);
+    return;
   }
+  State.getSema().Diag(Attr.getLoc(), diag::err_attribute_not_type_attr)
```

Comment:
The other error that says "only applies to parameters and implicit object parameters" seems to be provide more context.

Should we maybe use that instead? Are there any reasons to use the more generic one?

Reply:
SG, thanks!